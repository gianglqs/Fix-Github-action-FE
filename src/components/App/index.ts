export { default as AppTextField } from "./TextField";

export { default as AppHeader } from "./Header";
export * from "./SearchBar";
