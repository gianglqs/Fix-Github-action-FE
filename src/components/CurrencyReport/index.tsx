import { Button, Grid, Typography } from '@mui/material';
import LineChart from '../chart/Line';
import _ from 'lodash';
import React, { useEffect, useRef, useState } from 'react';
import { useDispatch } from 'react-redux';
import { formatDate } from '@/utils/formatCell';
import { useTranslation } from 'react-i18next';
//components
import Slider from '@mui/material/Slider';

const ChartView: React.FC<any> = ({
   chartData,
   index,
   tooltip,
   currentCurrency,
   handleCloseItem,
}) => {
   const { t } = useTranslation();
   const chartRef = useRef({} as any);
   const [scaleX, setScaleX] = useState([0, 0]);
   const [scaleY, setScaleY] = useState([0, 0]);
   const [scaleY1, setScaleY1] = useState([0, 0]);
   const [xBoundary, setXBoundary] = useState([0, 0]);
   const [yBoundary, setYBoundary] = useState([0, 0]);
   const [y1Boundary, setY1Boundary] = useState([0, 0]);

   const isJPY = chartData[index].title.includes('JPY');
   const yTitle =
      chartData[index]?.data?.datasets
         ?.map((data) => data.label)
         ?.filter((currency) => currency != 'JPY')
         ?.join(' ') || '';
   //get chart axis boundary after first render
   useEffect(() => {
      setXBoundary([chartRef.current.scales.x.min * 0.9, chartRef.current.scales.x.max * 1.1]);
      setYBoundary([chartRef.current.scales.y.min * 0.9, chartRef.current.scales.y.max * 1.1]);
      setScaleY([
         chartRef.current.scales.y.min * 0.9 * 100,
         chartRef.current.scales.y.max * 1.1 * 100,
      ]);

      setY1Boundary([chartRef.current.scales.y1.min, chartRef.current.scales.y1.max]);
      setScaleY1([chartRef.current.scales.y1?.min * 100, chartRef.current.scales.y1?.max * 100]);
   }, []);
   let chartItemScales = {
      y1: {
         title: {
            text: 'JPY',
            display: true,
         },
         position: 'right',
         grid: {
            drawOnChartArea: false, // only want the grid lines for one axis to show up
         },
         display: false,
      },
      y: {
         title: {
            text: yTitle,
            display: true,
         },
         ticks: {
            callback: function (value) {
               return Number(value.toFixed(6));
            },
         },
      },
      x: {
         title: {
            text: '',
            display: true,
         },
      },
   } as any;
   if (yBoundary[1] != 0) {
      chartItemScales.y.max = scaleY[1] / 100;
      chartItemScales.y.min = scaleY[0] / 100;
   }

   if (y1Boundary[1] != 0 && isJPY) {
      chartItemScales.y1.display = true;
      chartItemScales.y1.min = scaleY1[0] / 100;
      chartItemScales.y1.max = scaleY1[1] / 100;
   }
   return (
      <Grid
         container
         sx={{
            marginTop: 1,
            marginBottom: 1,
            marginLeft: 2,
            marginRight: 2,
            border: '0.5 px solid black',
            borderRadius: 5,
            boxShadow: 5,
         }}
      >
         <Button
            id={index}
            variant="contained"
            sx={{ width: 50, height: 20, margin: 1 }}
            onClick={(item) => handleCloseItem(item)}
         >
            Close
         </Button>

         <div
            id={`chart-${index}`}
            style={{
               width: '90%',
               height: 'fit-content',
               display: 'flex',
               alignItems: 'center',
               justifyContent: 'space-between',
            }}
         >
            <div
               style={{
                  display: 'flex',
                  flexDirection: 'column',
                  width: '100%',
                  justifyContent: 'center',
                  alignItems: 'center',
               }}
            >
               <div
                  style={{
                     display: 'flex',
                     height: '50vh',
                     gap: '10px',
                     width: '100%',
                     justifyContent: 'center',
                     alignItems: 'center',
                  }}
               >
                  <div
                     style={{
                        height: '50%',
                        order: 1,
                     }}
                  >
                     {yBoundary[1] && (
                        <Slider
                           max={yBoundary[1] * 100}
                           min={yBoundary[0] * 100}
                           value={scaleY}
                           onChange={(option, value) => {
                              setScaleY(value as number[]);
                           }}
                           orientation="vertical"
                        />
                     )}
                  </div>
                  <Grid
                     item
                     sx={{
                        width: '100%',
                        height: '50vh',
                        margin: 'auto',
                        order: 2,
                        flex: 1,
                     }}
                  >
                     <LineChart
                        ref={chartRef}
                        chartData={chartData[index].data}
                        chartName={chartData[index].title}
                        scales={chartItemScales}
                        tooltip={tooltip}
                        hover={{
                           mode: 'index',
                           intersect: false,
                        }}
                     />
                  </Grid>
                  <div
                     style={{
                        height: '50%',
                        order: 3,
                     }}
                  >
                     {y1Boundary[1] && isJPY && (
                        <Slider
                           max={y1Boundary[1] * 100}
                           min={y1Boundary[0] * 100}
                           value={scaleY1}
                           onChange={(option, value) => {
                              setScaleY1(value as number[]);
                           }}
                           orientation="vertical"
                        />
                     )}
                  </div>
               </div>
               {/*<div style={{ width: '60%' }}>
                  <Slider
                     max={100}
                     value={50}
                     onChange={(option, value) => {
                        setScaleX(value as number[]);
                     }}
                  />
               </div>*/}
            </div>
            <Grid
               item
               sx={{
                  width: '50vh',
                  height: 'fit-content',
               }}
            >
               <ul>
                  {chartData[index].conclusion.weakening.length > 0 ? (
                     <li>
                        <Typography fontSize={16}>
                           {currentCurrency} {t('exchangeRates.weakeningText')}
                        </Typography>{' '}
                        <ul>
                           {_.map(chartData[index].conclusion.weakening, (item) => (
                              <li>
                                 <Typography fontSize={14}>{item}</Typography>
                              </li>
                           ))}
                        </ul>
                     </li>
                  ) : (
                     ''
                  )}

                  {chartData[index].conclusion.stable.length > 0 ? (
                     <li>
                        <Typography fontSize={16}>
                           {currentCurrency} {t('exchangeRates.stableText')}
                           {_.map(
                              chartData[index].conclusion.stable,
                              (item, i) => `${i != '0' ? ', ' : ' '}${item}`
                           )}
                        </Typography>{' '}
                     </li>
                  ) : (
                     ''
                  )}

                  {chartData[index].conclusion.strengthening.length > 0 ? (
                     <li>
                        <Typography fontSize={16}>
                           {currentCurrency} {t('exchangeRates.strengtheningText')}
                        </Typography>{' '}
                        <ul>
                           {_.map(chartData[index].conclusion.strengthening, (item) => (
                              <li>
                                 {' '}
                                 <Typography fontSize={14}>{item}</Typography>
                              </li>
                           ))}
                        </ul>
                     </li>
                  ) : (
                     ''
                  )}
               </ul>
            </Grid>
         </div>
      </Grid>
   );
};
const CurrencyReport: React.FC<any> = (props) => {
   const { chartData, closeAReportItem, itemRef } = props;

   const handleCloseItem = (item) => {
      closeAReportItem(item.target.id);
   };

   const dispatch = useDispatch();
   const { t } = useTranslation();

   const reportContent = () => {
      const otherOptions = _.keysIn(chartData);

      const content = _.map(otherOptions, (index) => {
         const currentCurrency = chartData[index].title.substring(0, 3);
         const tooltip = {
            position: 'nearest',
            interaction: {
               intersect: false,
               mode: 'index',
            },
            callbacks: {
               label: (context) => {
                  let label = context.dataset.label || '';
                  if (label) {
                     label += ': ';
                  }
                  label += `${Number(context.parsed.y.toFixed(6))}`;

                  return label;
               },
            },
         };
         let chartItemScales = {
            y1: {
               title: {
                  text: 'JPY',
                  display: true,
               },
               position: 'right',
               grid: {
                  drawOnChartArea: false, // only want the grid lines for one axis to show up
               },
               display: false,
            },
            y: {
               ticks: {
                  callback: function (value) {
                     return Number(value.toFixed(6));
                  },
               },
            },
            x: {
               title: {
                  text: '',
                  display: true,
               },
            },
         };

         return (
            <>
               <ChartView
                  key={chartData[index].title}
                  index={index}
                  chartData={chartData}
                  chartItemScales={chartItemScales}
                  currentCurrency={currentCurrency}
                  tooltip={tooltip}
                  handleCloseItem={handleCloseItem}
               />
               {/*<Grid
               container
               sx={{
                  marginTop: 1,
                  marginBottom: 1,
                  marginLeft: 2,
                  marginRight: 2,
                  border: '0.5 px solid black',
                  borderRadius: 5,
                  boxShadow: 5,
               }}
            >
               <Button
                  id={index}
                  variant="contained"
                  sx={{ width: 50, height: 20, margin: 1 }}
                  onClick={(item) => handleCloseItem(item)}
               >
                  Close
               </Button>

               <div
                  id={`chart-${index}`}
                  style={{
                     width: '90%',
                     height: 'fit-content',
                     display: 'flex',
                     alignItems: 'center',
                     justifyContent: 'space-between',
                  }}
               >
                  <Grid
                     item
                     sx={{
                        width: '80vh',
                        height: '40vh',
                        margin: 'auto',
                     }}
                  >
                     <LineChart
                        chartData={chartData[index].data}
                        chartName={chartData[index].title}
                        scales={chartItemScales}
                        tooltip={tooltip}
                     />
                  </Grid>
                  <Grid
                     item
                     sx={{
                        width: '50vh',
                        height: 'fit-content',
                     }}
                  >
                     <ul>
                        {chartData[index].conclusion.weakening.length > 0 ? (
                           <li>
                              <Typography fontSize={16}>
                                 {currentCurrency} {t('exchangeRates.weakeningText')}
                              </Typography>{' '}
                              <ul>
                                 {_.map(chartData[index].conclusion.weakening, (item) => (
                                    <li>
                                       <Typography fontSize={14}>{item}</Typography>
                                    </li>
                                 ))}
                              </ul>
                           </li>
                        ) : (
                           ''
                        )}

                        {chartData[index].conclusion.stable.length > 0 ? (
                           <li>
                              <Typography fontSize={16}>
                                 {currentCurrency} {t('exchangeRates.stableText')}
                                 {_.map(
                                    chartData[index].conclusion.stable,
                                    (item, i) => `${i != '0' ? ', ' : ' '}${item}`
                                 )}
                              </Typography>{' '}
                           </li>
                        ) : (
                           ''
                        )}

                        {chartData[index].conclusion.strengthening.length > 0 ? (
                           <li>
                              <Typography fontSize={16}>
                                 {currentCurrency} {t('exchangeRates.strengtheningText')}
                              </Typography>{' '}
                              <ul>
                                 {_.map(chartData[index].conclusion.strengthening, (item) => (
                                    <li>
                                       {' '}
                                       <Typography fontSize={14}>{item}</Typography>
                                    </li>
                                 ))}
                              </ul>
                           </li>
                        ) : (
                           ''
                        )}
                     </ul>
                  </Grid>
               </div>
            </Grid>*/}
            </>
         );
      });

      return (
         <Grid
            container
            sx={{
               height: '80vh',
               overflowY: 'scroll',
            }}
            ref={itemRef}
         >
            {content}
         </Grid>
      );
   };
   return <>{reportContent()}</>;
};

export { CurrencyReport };
