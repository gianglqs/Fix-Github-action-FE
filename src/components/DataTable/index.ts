export { default as DataTable } from "./Main"

export * from './Pagination'
