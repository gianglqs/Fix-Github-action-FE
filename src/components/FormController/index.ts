export { default as FormControllerErrorMessage } from './ErrorMessage'