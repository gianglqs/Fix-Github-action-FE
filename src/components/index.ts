export * from "./FormController"
export * from "./Svg/Icon"
export * from "./DataTable"
export * from "./App"
