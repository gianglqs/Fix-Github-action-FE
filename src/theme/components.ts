import { red } from "@mui/material/colors"
import { secondaryColor } from "./palette"

import type { Components } from "@mui/material"
import type {} from "@mui/x-data-grid-pro/themeAugmentation"
import type {} from "@mui/lab/themeAugmentation"

const components: Components = {
  MuiButton: {
    defaultProps: {
      size: "small",
      disableElevation: true,
    },
    styleOverrides: {
      root: {
        textTransform: "capitalize",
      },
    },
  },
  MuiCheckbox: {
    defaultProps: {
      size: "small",
      color: "primary",
    },
  },
  MuiTextField: {
    defaultProps: {
      size: "small",
      variant: "outlined",
      fullWidth: true,
    },
    styleOverrides: {
      root: {
        "& input": {
          padding: "5px 8px !important",
          height: 14,
        },
      },
    },
  },
  MuiInputLabel: {
    styleOverrides: {
      root: {
        fontSize: 12,
        transform: "translate(14px, 2px) scale(1)",
      },
      shrink: {
        transform: "translate(14px, -9px) scale(0.8)",
        "& legend": {
          padding: 5,
        },
      },
    },
  },
  MuiTabs: {
    styleOverrides: {
      root: {
        minHeight: "38px !important",
        marginBottom: "12px !important",
      },
    },
  },
  MuiTab: {
    styleOverrides: {
      root: {
        minHeight: "32px !important",
        padding: "0 8px !important",
      },
    },
  },
  MuiInputBase: {
    styleOverrides: {
      multiline: {
        padding: "4px 8px !important",
      },
      adornedEnd: {
        paddingRight: "0 !important",
      },
    },
  },
  MuiTooltip: {
    defaultProps: {
      arrow: true,
    },
  },
  MuiFormLabel: {
    styleOverrides: {
      asterisk: {
        color: red[500],
      },
    },
  },
  MuiFormControl: {
    styleOverrides: {},
  },
  MuiChip: {
    defaultProps: {
      size: "small",
    },
  },
  MuiSvgIcon: {
    defaultProps: {
      fontSize: "small",
    },
    styleOverrides: {
      root: {
        fontSize: "1.115rem",
        cursor: "pointer !important",
      },
    },
  },
  MuiAutocomplete: {
    defaultProps: {
      openOnFocus: false,
    },
    styleOverrides: {
      inputRoot: {
        padding: "0 !important",
      },
      popupIndicator: {
        height: 26,
        padding: "0 !important",
      },
    },
  },
  MuiDialog: {
    styleOverrides: {
      paperWidthSm: {
        width: 550,
      },
    },
  },
  MuiDialogTitle: {
    styleOverrides: {
      root: {
        padding: "8px 16px",
      },
    },
  },
  MuiDialogActions: {
    styleOverrides: {
      root: {
        padding: "4px",
      },
    },
  },
  MuiSkeleton: {
    styleOverrides: {
      root: {
        backgroundColor: secondaryColor.light,
      },
    },
  },
  MuiDataGrid: {
    styleOverrides: {
      root: {
        border: "none",
        "& .MuiDataGrid-columnHeaders": {
          borderBottom: "none",
          backgroundColor: secondaryColor.main,
          color: "#606F89",
          fontWeight: "bold",
          lineHeight: "1.5rem",
        },
        "& .MuiDataGrid-row": {
          "&:hover": {
            backgroundColor: `${secondaryColor.light} !important`,
          },
          "&.Mui-selected": {
            backgroundColor: `${secondaryColor.light} !important`,
          },
        },
        "& .MuiDataGrid-row:nth-child(odd)": {
          backgroundColor: "#ffffff",
        },
        "& .MuiDataGrid-row:nth-child(even)": {
          backgroundColor: secondaryColor.main,
        },
        "& .MuiDataGrid-cell": {
          borderBottom: "none",
          color: secondaryColor.contrastText,
          "&:focus-within": {
            outline: "none",
          },
        },
        "& .MuiDataGrid-columnHeaderTitle": {
          fontWeight: "bold",
        },
        "& .MuiDataGrid-columnHeaderDraggableContainer": {
          "& .MuiDataGrid-menuIcon": {
            visibility: "visible !important",
            width: "auto",
          },
        },
      },
    },
  },
  MuiTreeItem: {
    styleOverrides: {
      content: {
        "&.Mui-focused": {
          backgroundColor: "white",
        },
        "&:hover": {
          backgroundColor: "white",
        },
      },
    },
  },
}

export default components
