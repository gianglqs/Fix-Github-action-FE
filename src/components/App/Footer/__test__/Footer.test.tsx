/*******************************************************
Copyright (C) - DREVER International

This file is part of Malis 3 project.

Malis 3 source code can not be copied and/or distributed without the express permission of DREVER International
*******************************************************/

import { renderTestUI } from '@/utils/testing/renderTestUI'
import { AppFooter } from '@/components'

import { RenderResult, cleanup } from '@testing-library/react'

import { FOOTER as footerHeight } from '@/styles/vars/size'

/*
 * TEST CASES
 * 1. Should be rendered without crashing
 * 2. Should be rendered a text: "© Copyright -y DREVER INTERNATIONAL" and -y is current year
 * 3. The text should be centered and height footer is equal footerHeight variable
 */

describe('<AppFooter />', () => {
  const year = new Date().getFullYear()
  const footerText = `© Copyright ${year} DREVER INTERNATIONAL`

  let wrapper: RenderResult

  beforeEach(() => {
    wrapper = renderTestUI(<AppFooter />)
  })
  afterEach(cleanup)

  test('should be rendered without crashing', () => {
    expect(wrapper.container).toMatchSnapshot()
  })

  test('should be rendered a text: "© Copyright -y DREVER INTERNATIONAL" and -y is current year', () => {
    expect(wrapper.getByTestId('footer-test')).toHaveTextContent(footerText)
  })

  test('The text should be centered and height to equal FOOTER variable', () => {
    expect(wrapper.getByTestId('footer-test')).toHaveStyle({
      height: `${footerHeight}px`,
      width: '100%',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center'
    })
  })
})
