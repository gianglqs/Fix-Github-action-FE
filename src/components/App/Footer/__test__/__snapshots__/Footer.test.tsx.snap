// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppFooter /> should be rendered without crashing 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 makeStyles-footer__container-1 css-15l6bmf-MuiPaper-root"
    data-testid="footer-test"
  >
    Â© Copyright 
    2022
     DREVER INTERNATIONAL
  </div>
</div>
`;
